core/domain/exp_domain.py

```diff
@@ -2384,8 +2366,31 @@ def from_yaml(cls, exploration_id, yaml_content):
         """
         exploration_dict = cls._migrate_to_latest_yaml_version(
             yaml_content, exploration_id)
+        exploration_dict = cls.add_id_and_proto_size_in_bytes_to_dict(
+            exploration_dict, exploration_id)
+
+        return Exploration.from_dict(exploration_dict)
+
+    def add_id_and_proto_size_in_bytes_to_dict(
+        cls, exploration_dict, exploration_id
+    ):
+        """Add id and proto_size_in_bytes to the exploration dict.
+
+        Args:
+            exploration_dict: dict. The dict representation of Exploration
+                object.
+            exploration_id: str. The exploration id.
+
+        Returns:
+            exploration_dict: dict. The updated dict representation of
+                Exploration object.
+        """
         exploration_dict['id'] = exploration_id
 
+        # Creating a exploration domain object to calcualte the
+        # proto size of it. If there is any discrepancy between
+        # the backend-computed size and the dict size then we need
+        # to move the proto_size_in_bytes argument to the constrcutor.
```

Comment:
I don't know what this last sentence means. What is the "dict size"? Maybe you can just say something like "If the proto_size_in_bytes of the dict is wrong, it will be overwritten with the correct value calculated from the exploration object" or something like that.

Btw, does the proto size calculation include the "proto size of the proto_size_in_bytes" field? If so then will we run into odd corner cases where some slight change in that field results in the proto size computation becoming off by 1?

Reply:
> does the proto size calculation include the "proto size of the proto_size_in_bytes" field?

No.